### PR TITLE
[FSSDK-9591] chore: bump go version to 1.21.0

### PR DESCRIPTION
**Pull request by pulak-opti**
## Ticket: 
[FSSDK-9591](https://jira.sso.episerver.net/browse/FSSDK-9591)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Refer to the [Go SDK's developer documentation](https://docs.developers.optimize
   
 ### Requirements  
 
-Requires Golang version 1.19 or higher.
+Requires Golang version 1.21.0 or higher.
```
Comment (mikechu-optimizely) [inline README.md:19]: Should we have a minimum of 1.20.7 (latest release of still supported version)?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions/setup-go@v3
       with:
-        go-version: '1.19'
+        go-version: '1.21.0'
```
Comment (mikechu-optimizely) [inline .github/workflows/go.yml:22]: Should we use a matrix strategy execution to test against 1.20.7 (last release of a supported version) at the same time?

**Review by mikechu-optimizely — APPROVED** [review]
Just a thought about 1.20, but otherwisse LGTM

File: .github/workflows/go.yml (hunk 2 above)
Comment (pulak-opti) [reply .github/workflows/go.yml:22]: When we start using the latest built-ins & packages of Go 1.21, tests will fail against anything lower than go version 1.21. Eventually, we can only test against the minimum required version of Go i.e. >= 1.21.

File: README.md (hunk 1 above)
Comment (pulak-opti) [reply README.md:19]: We can leave 1.19 for now. When we release the go-sdk with go 1.21, then this must be updated to 1.21.

**Pull request merged**